lib/matplotlib/tests/test_axes.py

```diff
@@ -2704,6 +2704,21 @@ def test_scatter_linewidths(self):
         assert_array_equal(pc.get_linewidths(),
                            [*range(1, 5), mpl.rcParams['lines.linewidth']])
 
+    def test_scatter_singular_plural_arguments(self):
+
+        with pytest.raises(TypeError, match="linewidths and linewidth cannot be"):
+            plt.scatter([1, 2, 3], [1, 2, 3], linewidths=[0.5, 0.4, 0.3], linewidth=0.2)
+
+        with pytest.raises(TypeError, match="edgecolors and edgecolor cannot be"):
+            plt.scatter([1, 2, 3], [1, 2, 3],
+                        edgecolors=["#ffffff", "#000000", "#f0f0f0"],
+                          edgecolor="#ffffff")
+
+        assert plt.scatter([1, 2, 3], [1, 2, 3], linewidths=[0.5, 0.4, 0.3],
+                           edgecolor="#ffffff")
+        assert plt.scatter([1, 2, 3], [1, 2, 3], linewidth=0.3,
+                           edgecolors=["#ffffff", "#000000", "#f0f0f0"])
```

Comment:
Not sure these are required as they basically checks that there are no errors in these cases? Or does it complain about not having full coverage if these are not there?

I think the errors now has nothing to do with your PR.

Reply:
Sorry that I didn't see this earlier.

Reply:
No problem. Should I remove the asserts?